contracts/exchange-forwarder/contracts/src/libs/LibForwarderRichErrors.sol

```diff
@@ -0,0 +1,214 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "@0x/contracts-utils/contracts/src/LibRichErrors.sol";
+
+
+library LibForwarderRichErrors {
+
+    // bytes4(keccak256("UnregisteredAssetProxyError()"))
+    bytes internal constant UNREGISTERED_ASSET_PROXY_ERROR =
+        hex"f3b96b8d";
+
+    // bytes4(keccak256("UnsupportedAssetProxyError(bytes4)"))
+    bytes4 internal constant UNSUPPORTED_ASSET_PROXY_ERROR_SELECTOR =
+        0x7996a271;
+
+    // bytes4(keccak256("CompleteFillFailedError()"))
+    bytes internal constant COMPLETE_FILL_FAILED_ERROR =
+        hex"4bcb2058";
+
+    // bytes4(keccak256("MakerAssetMismatchError(bytes,bytes)"))
+    bytes4 internal constant MAKER_ASSET_MISMATCH_ERROR_SELECTOR =
+        0x56677f2c;
+
+    // bytes4(keccak256("UnsupportedFeeError(bytes)"))
+    bytes4 internal constant UNSUPPORTED_FEE_ERROR_SELECTOR =
+        0x31360af1;
+
+    // bytes4(keccak256("FeePercentageTooLargeError(uint256)"))
+    bytes4 internal constant FEE_PERCENTAGE_TOO_LARGE_ERROR_SELECTOR =
+        0x1174fb80;
+
+    // bytes4(keccak256("InsufficientEthForFeeError()"))
+    bytes internal constant INSUFFICIENT_ETH_FOR_FEE_ERROR =
+        hex"54a53d9e";
+
+    // bytes4(keccak256("OversoldWethError(uint256,uint256)"))
+    bytes4 internal constant OVERSOLD_WETH_ERROR_SELECTOR =
+        0x5cc555c8;
+
+    // bytes4(keccak256("TransferFailedError()"))
+    bytes internal constant TRANSFER_FAILED_ERROR =
+        hex"570f1df4";
+
+    // bytes4(keccak256("DefaultFunctionWethContractOnlyError(address)"))
+    bytes4 internal constant DEFAULT_FUNCTION_WETH_CONTRACT_ONLY_ERROR_SELECTOR =
+        0x08b18698;
+
+    // bytes4(keccak256("InvalidMsgValueError()"))
+    bytes4 internal constant INVALID_MSG_VALUE_ERROR_SELECTOR =
+        0xb0658a43;
+
+    // bytes4(keccak256("InvalidMsgValueError()"))
+    bytes internal constant INVALID_MSG_VALUE_ERROR =
+        hex"b0658a43";
+
+    // bytes4(keccak256("InvalidErc721AmountError(uint256)"))
+    bytes4 internal constant INVALID_ERC721_AMOUNT_ERROR_SELECTOR =
+        0x27ed87bf;
+
+    // solhint-disable func-name-mixedcase
+    function UnregisteredAssetProxyError()
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return UNREGISTERED_ASSET_PROXY_ERROR;
+    }
+
+    function UnsupportedAssetProxyError(
+        bytes4 proxyId
+    )
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodeWithSelector(
+            UNSUPPORTED_ASSET_PROXY_ERROR_SELECTOR,
+            proxyId
+        );
+    }
+
+    function CompleteFillFailedError()
```

Comment:
We have a similar error in the `Exchange` contract and recently added `uint256 expectedAssetFillAmount` and `uint256 actualAssetFillAmount` as parameters. Can we make that change here too?